h5utils: init at 1.13.1

SFrijters opened the pull request:
###### Motivation for this change

New package: "A set of utilities for visualization and conversion of scientific data in the free, portable HDF5 format"

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

SFrijters committed:
h5utils: init at 1.13.1

nlewo commented on pkgs/tools/misc/h5utils/default.nix:
```diff
@@ -0,0 +1,26 @@
+  version = "1.13";
+  name = "h5utils-${version}";
+
+  src = fetchurl {
```
Could you please switch to `fetchFromGitHub`.

nlewo replied:
Ok. I also think it's simpler to fetch this archive. So `fetchurl` is fine in this case.
Could you then just add a comment explaining this choice?

nlewo commented on pkgs/tools/misc/h5utils/default.nix:
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl, hdf5, lib }:
+
+let
+  mpi = hdf5.mpi;
```
(non blocking) This was not helping me to read your code. So from my point of view, this could be removed.

nlewo commented on pkgs/tools/misc/h5utils/default.nix:
```diff
@@ -0,0 +1,26 @@
+    sha256 = "1574wrymvcp3zh1m57f2c2ajxhbvpkfb04hyx1am95q856a0b2vy";
+  };
+
+  preBuild = if mpiSupport then "export CC=${mpi}/bin/mpicc" else "";
```
tips: you could use `lib.optionalString`

SFrijters commented:
I've implemented @nlewo's suggestions, but I've also extended the derivation with some optional dependencies so it could use another look; it seemed to me that different packages use slightly different ways to do this (nullable dependencies, or boolean flags), so please let me know if I should change the way I did it here.

Also, it seems that the tarball release currently does not include some build fixes needed for some of the optional dependencies, so I've asked the upstream author to cut a new release if possible. If no response is forthcoming I could switch to fetchFromGitHub after all.

nlewo commented:
Ok, great. 
On my side, this PR could be merged. But it seems you would prefer to see some fixes integrated.
So, let's wait some days to see if a new release is coming. Ping me when you want to move forward on this PR.

SFrijters commented:
What is the policy on using arbitrary git revisions? I have a working derivation that uses current git master and autoreconfHook to build without problems, but just like the tarball, the latest git tag is 1.13 and that one lacks some build fixes. What I have now is:

```
  # The revision below is release 1.13 plus various fixes
  # needed to build with the optional dependencies
  src = fetchFromGitHub {
    owner = "stevengj";
    repo = "h5utils";
    rev = "e17b1354b20270a3920a8c61bee8c412c0429eae";
    sha256 = "1fwwsy4kar1jz31jax2ivzfaj8ldc0j77x1z0694i0lj69mj6f1a";
  };
```

SFrijters commented:
@nlewo The package author has made a 1.13.1 point release, so my previous build errors are now fixed.

I did run into another issue though: with the optional `octave` dependency, a plugin is built that is copied into the `octave` directory in the Nix store, which obviously is read-only. It doesn't look like plugins are supported by the octave derivation, so I've removed this optional dependency for now, unless you think there is a way to handle this gracefully?

nlewo commented:
@SFrijters ok, that's cool for the new release. I know nothing regarding `octave` so I could not really help you on this subject. This PR looks good to me.
@GrahamcOfBorg build h5utils

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.37092&attempt_id=4bc7a2f4-607f-401c-8316-30cd26921b93)

Attempted: h5utils

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1/bin/h5fromtxt
shrinking /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1/bin/h5tovtk
shrinking /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1/bin/h5totxt
shrinking /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1/bin/h5topng
gzipping man pages under /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1/share/man/
strip is /nix/store/fzcs0fn6bb04m82frhlb78nc03ny3w55-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1/bin
patching script interpreter paths in /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1
checking for references to /build in /nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1...
/nix/store/am89gz8wm82srxxxawm53qblksw3r8hp-h5utils-1.13.1
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.37092&attempt_id=35c3b638-9c9f-41d2-bc94-91dfb56c186e)

Attempted: h5utils

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1/bin/h5topng
shrinking /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1/bin/h5tovtk
shrinking /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1/bin/h5fromtxt
shrinking /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1/bin/h5totxt
gzipping man pages under /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1/share/man/
strip is /nix/store/3zq400fri5dv7d30lpxlqm2v9y1iis6j-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1/bin
patching script interpreter paths in /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1
checking for references to /build in /nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1...
/nix/store/7dwsm8hv6x2zngh2y17899xm4yrdqhyd-h5utils-1.13.1
```
</p></details>

nlewo commented:
Thanks!

SFrijters commented:
Thanks for the coaching!
